app/models/case_record.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require "securerandom"
+
+class CaseRecord < ApplicationRecord
+  delegated_type :caseable, types: %w[Enforcement], dependent: :destroy
+
+  after_initialize :generate_uuid_v7
```

Comment:
I'd leave the `_v7` off as that's an implementation detail of the private method